src/android/aidl/birthday_service/src/client.rs

```diff
@@ -34,8 +34,65 @@ fn main() -> Result<(), binder::Status> {
         .unwrap_or(42);
 
     binder::ProcessState::start_thread_pool();
-    let service = connect().expect("Failed to connect to BirthdayService");
+    let service: binder::Strong<dyn IBirthdayService> =
+        binder::get_interface(SERVICE_IDENTIFIER)
+            .expect("Failed to connect to BirthdayService");
+
+    // Call the service.
     let msg = service.wishHappyBirthday(&name, years)?;
     println!("{msg}");
+    // ANCHOR_END: main
+
+    // ANCHOR: wish_with_info
+    service.wishWithInfo(&BirthdayInfo { name: name.clone(), years })?;
+    // ANCHOR_END: wish_with_info
+
+    // ANCHOR: wish_with_provider
+
+    // Create a binder object for the `IBirthdayInfoProvider` interface.
+    let provider = BnBirthdayInfoProvider::new_binder(
+        InfoProvider { name: name.clone(), age: years as u8 },
+        BinderFeatures::default(),
+    );
+
+    // Send the binder object to the service.
+    service.wishWithProvider(&provider)?;
+
+    // Perform the same operation but passing the provider as an `SpIBinder`.
+    service.wishWithErasedProvider(&provider.as_binder())?;
+    // ANCHOR_END: wish_with_provider
+
+    // ANCHOR: wish_with_file
+
+    // Open a file and put the birthday info in it.
+    let mut file = File::create("/data/local/tmp/birthday.info").unwrap();
+    writeln!(file, "{name}").unwrap();
+    writeln!(file, "{years}").unwrap();
+
+    // Create a `ParcelFileDescriptor` from the file and send it.
+    let file = ParcelFileDescriptor::new(file);
```

Comment:
Do you need `file.into()` here? We switched PFD to take an `OwnedFd` recently.

Reply:
Doesn't seem like it? I'm able to `repo sync` and build against latest without the `.into()`. Would those changes be on a particular AOSP branch? I'm on whatever main branch gets checked out by default.

Reply:
If you're on `main`  and  you tested the current code there, that's good enough for me.